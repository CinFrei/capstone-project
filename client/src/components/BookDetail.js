import PropTypes from 'prop-types'
import styled from 'styled-components/macro'

import RectangleButton from './RectangleButton'

/** @component */
export default BookDetail

BookDetail.propTypes = {
  buttonName: PropTypes.string,
  detailBook: PropTypes.array,
  onButtonClick: PropTypes.func,
}

function BookDetail({ buttonName, detailBook, onButtonClick }) {
  return (
    <BookDetailModal>
      {detailBook.map((detailBook) => (
        <DetailWrapperStyled key={detailBook.id}>
          <CoverStyled
            src={detailBook.volumeInfo.imageLinks.thumbnail}
            alt={detailBook.volumeInfo.title}
          />
          <DetailsStyled>
            <h2>{detailBook.volumeInfo.title}</h2>
            <h3>{detailBook.volumeInfo.subtitle}</h3>
            <table>
              <tr>
                <th>Autor: </th>
                <th> {detailBook.volumeInfo.authors}</th>
              </tr>
              <tr>
                <th>Kathegorien: </th>
                <th> {detailBook.volumeInfo.categories}</th>
              </tr>
              <tr>
                <th>Seiten: </th>
                <th> {detailBook.volumeInfo.pageCount}</th>
              </tr>
              <tr>
                <th>Format: </th>
                <th> {detailBook.volumeInfo.printType}</th>
              </tr>
              <tr>
                <th>
                  Erscheinungs-
                  <br />
                  datum:
                </th>
                <th> {detailBook.volumeInfo.publishedDate}</th>
              </tr>
              <tr>
                <th>Verlag: </th>
                <th> {detailBook.volumeInfo.publisher}</th>
              </tr>
              <tr>
                <th>Sprache: </th>
                <th> {detailBook.volumeInfo.language}</th>
              </tr>
            </table>
            <p>{detailBook.volumeInfo.description}</p>
          </DetailsStyled>
        </DetailWrapperStyled>
      ))}
      <RectangleButton
        buttonName={buttonName}
        gridRow="2"
        gridPosition="end"
        onClick={onButtonClick}
      />
    </BookDetailModal>
  )
}

const BookDetailModal = styled.section`
  background: var(--dark-blue);
  display: grid;
  grid-gap: 10px;
  grid-template-rows: auto 50px;
  height: 100%;
  min-width: 0;
  overflow-y: scroll;
  place-items: center;
  position: fixed;
  right: 0;
  top: 0px;
  width: 100%;
  z-index: 20;

  h2 {
    margin-top: 20px;
  }
  h3 {
    margin: 5px 0 20px;
  }

  p {
    margin: 15px 5px 0 3px;
  }

  button {
    bottom: 20px;
    position: fixed;
    right: 5%;
    width: 90%;
    z-index: 10;
  }
`

const CoverStyled = styled.img`
  grid-row: 1;
  position: fixed;
  right: 0;
  top: 0;
  width: 100%;
`

const DetailsStyled = styled.div`
  background: var(--dark-blue-gradient);
  grid-row: 2;
  height: 100%;
  padding: 110px 60px;
  z-index: 2;
`

const DetailWrapperStyled = styled.div`
  display: grid;
  grid-gap: 10px;
  grid-row: 1;
  grid-template-rows: 25% auto;
  height: 100%;

  table {
    text-align: left;
  }
`
