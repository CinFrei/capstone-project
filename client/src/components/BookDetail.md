BookDetail:

```js
<BookDetail
  author="Andrzej Sapkowski"
  buttonName="zurück"
  categories="Fiction"
  cover="./public/styleguide/sample-cover.jpg"
  description="Der Hexer ist wieder auf einer Spannenden Reise, auf der er Monster besiegt, Prinzessin Ciri rettet, mit Dryaden spricht und flotte Sprüche abgibt."
  language="DE"
  pageCount="564"
  printType="Book"
  publishedDate="12.12.1992"
  publisher="Hüpfer Verlag"
  subtitle="Der Witcher"
  title="Das Schwert der Vorsehung"
/>
```
