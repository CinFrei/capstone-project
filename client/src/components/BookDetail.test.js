import BookDetail from './BookDetail'
import { render } from '@testing-library/react'

describe('BookDetail', () => {
  it('renders correctly', () => {
    const { container } = render(
      <BookDetail
        detailBook={[
          {
            volumeInfo: {
              authors: [`Paul Tobin`],
              imageLinks: {
                thumbnail:
                  'http://books.google.com/books/content?id=6D13DwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api',
              },
              title: 'The Witcher, Band 2 - Fuchskinder',
              categories: 'Fiction',
              pageCount: '564',
              printType: 'Book',
              publishedDate: '12.12.1992',
              publisher: 'Hüpfer Verlag',
              language: 'DE',
              description:
                'Der Hexer ist wieder auf einer Spannenden Reise, auf der er Monster besiegt, Prinzessin Ciri rettet, mit Dryaden spricht und flotte Sprüche abgibt.',
              subtitle: 'Der Witcher',
            },
          },
        ]}
        buttonName="Buch hinzufügen."
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
