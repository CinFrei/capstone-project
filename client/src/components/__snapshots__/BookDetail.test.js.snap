// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookDetail renders correctly 1`] = `
.c4 {
  background: var(--accent-color);
  border-radius: 3px;
  border: none;
  color: var(--primary-anthrazit);
  font-family: 'Kaushan Script',cursive;
  font-size: 1.2em;
  grid-row: 2;
  justify-self: end;
  width: 100%;
}

.c0 {
  background: var(--dark-blue);
  display: grid;
  grid-gap: 10px;
  grid-template-rows: auto 50px;
  height: 100%;
  min-width: 0;
  overflow-y: scroll;
  place-items: center;
  position: fixed;
  right: 0;
  top: 0px;
  width: 100%;
  z-index: 20;
}

.c0 h2 {
  margin: 20px 0 0;
}

.c0 h3 {
  margin: 5px 0 20px;
}

.c0 span {
  margin: 0 0 0 3.5px;
}

.c0 p {
  margin: 15px 5px 0 3px;
}

.c0 button {
  position: fixed;
  right: 5%;
  width: 90%;
  z-index: 10;
  bottom: 20px;
}

.c2 {
  grid-row: 1;
  width: 100%;
  position: fixed;
  top: 0;
  right: 0;
}

.c3 {
  grid-row: 2;
  height: 100%;
  z-index: 2;
  background: var(--dark-blue-gradient);
  padding: 110px 60px;
}

.c1 {
  grid-row: 1;
  display: grid;
  grid-gap: 10px;
  grid-template-rows: 25% auto;
  height: 100%;
}

.c1 table {
  text-align: left;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <img
      alt="The Witcher, Band 2 - Fuchskinder"
      class="c2"
      src="http://books.google.com/books/content?id=6D13DwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
    />
    <div
      class="c3"
    >
      <h2>
        The Witcher, Band 2 - Fuchskinder
      </h2>
      <h3>
        Der Witcher
      </h3>
      <table>
        <tr>
          <th>
            Autor: 
          </th>
          <th>
             
            Paul Tobin
          </th>
        </tr>
        <tr>
          <th>
            Kathegorien: 
          </th>
          <th>
             
            Fiction
          </th>
        </tr>
        <tr>
          <th>
            Seiten: 
          </th>
          <th>
             
            564
          </th>
        </tr>
        <tr>
          <th>
            Format: 
          </th>
          <th>
             
            Book
          </th>
        </tr>
        <tr>
          <th>
            Erscheinungs-
            <br />
            datum:
             
          </th>
          <th>
             
            12.12.1992
          </th>
        </tr>
        <tr>
          <th>
            Verlag: 
          </th>
          <th>
             
            Hüpfer Verlag
          </th>
        </tr>
        <tr>
          <th>
            Sprache: 
          </th>
          <th>
             
            DE
          </th>
        </tr>
      </table>
      <p>
        Der Hexer ist wieder auf einer Spannenden Reise, auf der er Monster besiegt, Prinzessin Ciri rettet, mit Dryaden spricht und flotte Sprüche abgibt.
      </p>
    </div>
  </div>
  <button
    class="c4"
  >
    Buch hinzufügen.
  </button>
</div>
`;
